token/cli/src/main.rs

```diff
@@ -2738,6 +2748,120 @@ async fn command_withdraw_withheld_tokens(
     Ok(results.join(""))
 }
 
+async fn command_update_confidential_transfer_settings(
+    config: &Config<'_>,
+    token_pubkey: Pubkey,
+    authority: Pubkey,
+    auto_approve: Option<bool>,
+    auditor_pubkey: Option<ElGamalPubkeyOrNone>,
+    bulk_signers: Vec<Arc<dyn Signer>>,
+) -> CommandResult {
+    let (new_auto_approve, new_auditor_pubkey) = if !config.sign_only {
+        let confidential_transfer_account = config.get_account_checked(&token_pubkey).await?;
+
+        let mint_state =
+            StateWithExtensionsOwned::<Mint>::unpack(confidential_transfer_account.data)
+                .map_err(|_| format!("Could not deserialize token mint {}", token_pubkey))?;
+
+        if let Ok(confidential_transfer_mint) =
+            mint_state.get_extension::<ConfidentialTransferMint>()
+        {
+            let expected_authority = Option::<Pubkey>::from(confidential_transfer_mint.authority);
+
+            if expected_authority != Some(authority) {
+                return Err(format!(
+                    "Mint {} has confidential transfer authority {}, but {} was provided",
+                    token_pubkey,
+                    expected_authority
+                        .map(|pubkey| pubkey.to_string())
+                        .unwrap_or_else(|| "disabled".to_string()),
+                    authority
+                )
+                .into());
+            }
+
+            let current_auto_approve = if let Some(auto_approve) = auto_approve {
+                auto_approve
+            } else {
+                bool::from(confidential_transfer_mint.auto_approve_new_accounts)
+            };
+
+            let current_auditor_pubkey = if let Some(auditor_pubkey) = auditor_pubkey {
```

Comment:
micro-nit: same here
```suggestion
            let new_auditor_pubkey = if let Some(auditor_pubkey) = auditor_pubkey {
```